#1 - Allow multiple idp's with the same metadatalink

Anders-Oerbaek-Pedersen opened the pull request:
Fix for #1 

Notable changes:

* Changed use of internal array key by from non unique database field `entityid` to unique database field `id`

Anders-Oerbaek-Pedersen committed:
#1 - Changed keyby for idp's to use internal id instead of entryid

Entryid is not unique. Could result in same idp's entryid would only show last.

Anders-Oerbaek-Pedersen committed:
#1 - Changed plugin version

frederikmillingpytlick commented on classes/auth.php:
```diff
@@ -123,12 +123,13 @@ public function __construct() {
             // Set name.
             $idpentity->name = empty($idpentity->displayname) ? $idpentity->defaultname : $idpentity->displayname;
-            $idpentity->md5entityid = md5($idpentity->entityid);
+            $idpentity->md5id = md5($idpentity->entityid);
+            $idpentity->md5id = md5($idpentity->id);
```
Overwriting immediately after?

Anders-Oerbaek-Pedersen replied:
Forgot to cleanup original line, ended up changing that one also, when renaming object key.

I have removed the superfluous line.

Anders-Oerbaek-Pedersen committed:
#1 - Removed line accidentially left after initial changes

Anders-Oerbaek-Pedersen committed:
#1 - Changed use of entityid as array key to id

Oversight on inital changes, resulting in rows not being deleted, as expected when updating idp's

Logiar commented:
When saving on /admin/settings.php?section=authsettingsaml2 it does not map them to the old rows, causing new rows to be inserted and old ones to get wiped. This causes a reset to default for settings stored in auth_saml2_idps table. [classes/admin/setting_idpmetadata.php:158](https://github.com/praxisdigital/moodle-auth_saml2/blob/ccf0f4391863837202182e1fb37a87d3af38c99f/classes/admin/setting_idpmetadata.php#L158) is inconsistent with [classes/admin/setting_idpmetadata.php:93](https://github.com/praxisdigital/moodle-auth_saml2/blob/ccf0f4391863837202182e1fb37a87d3af38c99f/classes/admin/setting_idpmetadata.php#L93) and database id can't be used for mapping here.

Anders-Oerbaek-Pedersen committed:
#1 - Fixed issue with IdP matching when updating settings

Anders-Oerbaek-Pedersen committed:
#1 - bump plugin version

Anders-Oerbaek-Pedersen commented:
Mapping should be fixed now.
If the override name is changed, mapping will not match with the previous, causing the old row to be removed and a new row to be inserted.
This is caused by having nothing unique to map to, when allowing two identical IdP metadatalinks and IdP entries being declared as one text input.

Anders-Oerbaek-Pedersen commented:
Tested via show and tell with: JRM
